assets/blocks/reader-registration/edit.js

```diff
@@ -12,30 +21,197 @@ import './editor.scss';
 
 export default function ReaderRegistrationEdit( {
 	setAttributes,
-	attributes: { placeholder, label },
+	attributes: {
+		placeholder,
+		label,
+		privacyLabel,
+		newsletterSubscription,
+		displayListDescription,
+		newsletterTitle,
+		newsletterLabel,
+		lists,
+		className,
+	},
 } ) {
 	const blockProps = useBlockProps();
+	const [ inFlight, setInFlight ] = useState( false );
+	const [ listConfig, setListConfig ] = useState( {} );
+
+	const fetchLists = () => {
+		if ( newspack_blocks.has_newsletters && newsletterSubscription ) {
+			setInFlight( true );
+			apiFetch( {
+				path: '/newspack-newsletters/v1/lists_config',
+			} )
+				.then( setListConfig )
+				.finally( () => setInFlight( false ) );
+		}
+	};
+
+	useEffect( fetchLists, [] );
+	useEffect( fetchLists, [ newsletterSubscription ] );
+
+	useEffect( () => {
+		const listIds = Object.keys( listConfig );
+		if ( listIds.length && ( ! lists.length || ! intersection( lists, listIds ).length ) ) {
+			setAttributes( { lists: [ Object.keys( listConfig )[ 0 ] ] } );
+		}
+	}, [ listConfig ] );
+
 	return (
 		<>
 			<InspectorControls>
 				<PanelBody title={ __( 'Form settings', 'newspack' ) }>
 					<TextControl
 						label={ __( 'Input placeholder', 'newspack' ) }
 						value={ placeholder }
+						disabled={ inFlight }
 						onChange={ value => setAttributes( { placeholder: value } ) }
 					/>
 					<TextControl
 						label={ __( 'Button label', 'newspack' ) }
 						value={ label }
+						disabled={ inFlight }
 						onChange={ value => setAttributes( { label: value } ) }
 					/>
 				</PanelBody>
+				{ newspack_blocks.has_newsletters && (
+					<PanelBody title={ __( 'Newsletter Subscription', 'newspack' ) }>
+						<ToggleControl
+							label={ __( 'Enable newsletter subscription', 'newspack' ) }
+							checked={ !! newsletterSubscription }
+							disabled={ inFlight }
+							onChange={ () =>
+								setAttributes( { newsletterSubscription: ! newsletterSubscription } )
+							}
+						/>
+						{ newsletterSubscription && (
+							<>
+								{ inFlight && <Spinner /> }
+								{ ! inFlight && ! Object.keys( listConfig ).length && (
+									<div style={ { marginBottom: '1.5rem' } }>
+										<Notice isDismissible={ false } status="error">
+											{ __( 'You must enable lists for subscription.', 'newspack-newsletters' ) }
+										</Notice>
+									</div>
+								) }
+								{ lists.length === 1 && (
```

Comment:
Non-blocking: I'm not sure we need to have the editable fields in both the WSYIWYG editor and the sidebar.

Reply:
I agree! 155d24a06759eabc3e6c50ceacdc07953acb21ff removes the duplicate controls from the panel.

Which overrides are you referring to? If there's one list selected, its label should no longer be the list title and be a more direct generic newsletter subscription message. The same for the editable "Newsletters" title, which should only be available in the column layout and multiple newsletters.

Reply:
The editable label + description within the block is what I'm calling the "overrides", because they override the list name and description you set on the list itself in the Engagement wizard. I get the logic now, but I anticipate some initial publisher confusion about this.